compiler/src/dotty/tools/dotc/transform/PCPCheckAndHeal.scala

```diff
@@ -144,7 +144,17 @@ class PCPCheckAndHeal(@constructorOnly ictx: Context) extends TreeMapWithStages(
     /** Is a reference to a class but not `this.type` */
     def isClassRef = sym.isClass && !tp.isInstanceOf[ThisType]
 
-    if (!sym.exists || levelOK(sym))
+    /** Is this a static path or a type porjection with a static prefix */
+    def isStaticPathOK(tp1: Type): Boolean = tp1 match {
+      case tp1: TypeRef => tp1.symbol.is(Package) || isStaticPathOK(tp1.prefix)
+      case tp1: TermRef => tp1.symbol.isStaticOwner || isStaticPathOK(tp1.prefix)
+      case tp1: ThisType => tp1.tref.symbol.is(Package) || tp1.tref.symbol.is(Module)
+      case tp1: AppliedType => isStaticPathOK(tp1.tycon)
```

Comment:
Applied types appear with type projections such as in https://github.com/lampepfl/dotty/blob/master/tests/pos/i5954c.scala#L9